modules/raft/src/integrationTest/java/org/apache/ignite/internal/raft/ItTruncateSuffixAndRestartTest.java

```diff
@@ -0,0 +1,384 @@
+import org.apache.ignite.network.ClusterService;
+import org.apache.ignite.network.NettyBootstrapFactory;
+import org.apache.ignite.network.scalecube.TestScaleCubeClusterServiceFactory;
+import org.apache.ignite.raft.TestWriteCommand;
+import org.apache.ignite.raft.jraft.conf.Configuration;
+import org.apache.ignite.raft.jraft.conf.ConfigurationEntry;
+import org.apache.ignite.raft.jraft.conf.ConfigurationManager;
+import org.apache.ignite.raft.jraft.entity.EnumOutter;
+import org.apache.ignite.raft.jraft.entity.LogEntry;
+import org.apache.ignite.raft.jraft.entity.LogId;
+import org.apache.ignite.raft.jraft.option.LogStorageOptions;
+import org.apache.ignite.raft.jraft.option.RaftOptions;
+import org.apache.ignite.raft.jraft.storage.LogStorage;
+import org.apache.ignite.raft.jraft.storage.impl.OnHeapLogs;
+import org.apache.ignite.raft.jraft.storage.impl.UnlimitedBudget;
+import org.apache.ignite.raft.jraft.storage.impl.VolatileLogStorage;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+/**
+ * Test that checks various scenarios where raft node re-enter the group with lost data. Data loss may occur in case when
+ * "sync" is disabled, and last portion of the RAFT log wasn't stored durably due to power outage, for example.
+ */
+@ExtendWith(WorkDirectoryExtension.class)
+@ExtendWith(ConfigurationExtension.class)
+public class ItTruncateSuffixAndRestartTest {
+    private static final int NODES = 3;
+
+    private static final String GROUP_NAME = "foo";
+
+    private static final ReplicationGroupId GROUP_ID = new TestReplicationGroupId(GROUP_NAME);
+
+    private final PeersAndLearners raftGroupConfiguration = fromConsistentIds(
+            range(0, NODES).mapToObj(ItTruncateSuffixAndRestartTest::nodeName).collect(toSet())
+    );
+
+    private final HybridClock hybridClock = new HybridClockImpl();
+
+    /** List of resources to be released after each test. Greatly simplifies the code. */
+    private final List<ManuallyCloseable> cleanup = new ArrayList<>();
+
+    @WorkDirectory
+    private Path workDir;
+
+    @InjectConfiguration
+    private RaftConfiguration raftConfiguration;
+
+    @InjectConfiguration
+    private NetworkConfiguration networkConfiguration;
+
+    private List<SimpleIgniteNode> nodes;
+
+    @BeforeEach
+    void setUp() {
+        CompletableFuture<Void> changeFuture = networkConfiguration.change(cfg -> cfg
+                .changePortRange(0)
+                .changeNodeFinder().changeNetClusterNodes(
+                        range(port(0), port(NODES)).mapToObj(port -> "localhost:" + port).toArray(String[]::new)
+                )
+        );
+
+        assertThat(changeFuture, willCompleteSuccessfully());
+
+        nodes = range(0, NODES).mapToObj(SimpleIgniteNode::new).collect(toList());
+
+        nodes.forEach(SimpleIgniteNode::startService);
```

Comment:
Shouldn't it be added to `cleanup`?
Can we wait for all nodes to start?

Reply:
It's actually already added in the SimpleIgniteNode constructor